ambry-account/src/main/java/com/github/ambry/account/mysql/DatasetDao.java

```diff
@@ -557,18 +562,21 @@ public synchronized Dataset getDataset(int accountId, int containerId, String ac
    * @throws SQLException
    */
   public synchronized List<DatasetVersionRecord> getAllValidVersionForDatasetDeletion(short accountId,
-      short containerId, String datasetName) throws SQLException {
+      short containerId, String datasetName) throws SQLException, AccountServiceException {
     try {
       long startTimeMs = System.currentTimeMillis();
       dataAccessor.getDatabaseConnection(false);
+      PreparedStatement getVersionSchemaStatement = dataAccessor.getPreparedStatement(getVersionSchemaSql, false);
+      Dataset.VersionSchema versionSchema =
+          executeGetVersionSchema(getVersionSchemaStatement, accountId, containerId, datasetName);
       PreparedStatement listAllValidVersionForDatasetDeletionStatement =
           dataAccessor.getPreparedStatement(listValidVersionForDatasetDeletionSql, false);
       List<DatasetVersionRecord> datasetVersionRecords =
           executeListAllValidVersionForDatasetDeletionStatement(listAllValidVersionForDatasetDeletionStatement,
-              accountId, containerId, datasetName);
+              accountId, containerId, datasetName, versionSchema);
```

Comment:
Looks like right now we always serialize the version based on the versionSchema. Because of that, for each API, we need two mysql API calls. One is to get the versionSchema. The other is the real one which lists the version set either for deletion or for listAll.
Not sure if the extra getSchema query will have any performance impact. For long term, probably we can think about a way to avoid the getSchema mysql call. For example, probably client has this information already and it can provide to the API.

Reply:
the version schema is provided at dataset level(one time) before user adding the dataset version, so they don't need to provide the version schema during dataset version upload.

Reply:
Yes, I understand the dataset version schema is pre-determined by customer and it is saved in dataset table, not the version table.

Just for this API, in order to serialize the version, we have to make two mysql calls when we really care is the version table. I don't know if we have any latency data, for example how long is one mysql call takes and how long   we usually finish this API. Does the getSchema mysql call have any performance impact?